### PR TITLE
doc: Start documenting pseudomodules

### PR DESCRIPTION
**Pull request by chrysn**
### Contribution description

This adds a module documentation group to pseudomodules, and populates a few first entries that would have been needed in #17132.

### Testing procedure

* Look at documentation

### Caveats

* Makefiles are treated like Python files; this seems to be the only way to get Doxygen to look at them and not ignore the Makefile comments because they look like C directives. This could be prettier if Doxygen's [extension mappings](https://www.doxygen.nl/manual/starting.html#default_file_extension_mapping) were extensible beyond picking one of the known languages.
* A dedicated group is created for each module. This is a bit ugly as it's many small pages, and pages without content practically always repeat their brief line.

I think these are acceptable as long as we're using Doxygen.

### Issues/PRs references

The last commit soft-depends on #17132 which initially documents SCHED_TEST_STACK -- builds without, but CI may cry foul.

### CODE REVIEW

File: doc.txt [hunk 1]
```diff
@@ -34,7 +34,8 @@
  * The following list of what `DEVELHELP=1` enables is not comprehensive, but
  * should give a rough impression of what to expect:
  *
- * * Many runtime checks are enabled (stack overflow protection, warnings when
+ * * Many runtime checks are enabled (stack overflow protection by means of
+ *   @ref pseudomodule_mpu_stack_guard or @ref SCHED_TEST_STACK, warnings when
```
Comment (fjmolinas) [inline doc.txt:38]: Hmm for some reason `SCHED_TEST_STACK` not defined, I think the `#if defined(DEVELHELP) || defined(DOXYGEN)` needs to be only around the define and not the comment, can you try that?

**Review by fjmolinas — COMMENTED** [review]
Lets get this in to get the ball rolling on documentation!

File: doc.txt (hunk 1 above)
Comment (chrysn) [reply doc.txt:38]: Indeed; that was fixed in https://github.com/RIOT-OS/RIOT/pull/17132, and the hardships we had finding it was one of the things that triggered this.

I've applied @asbadie's suggestion, and will squash-rebase onto master to get the defined SCHED_TEST_STACK in.

**Review by fjmolinas — APPROVED** [review]
ACK! Good to have some documentation on this!

**Comment by chrysn** [comment]
Force-pushed without modifications to get a CircleCI build of the documentation, to verify that it looks as nice in actual builds as it looks locally.

**Comment by chrysn** [comment]
Having used pseudomodules now a bit more, I consider some blanket statements in the new documentation excessive in retrospect. (Also, I'm no longer convinced *all* pseudomodules should be documented here).

I've added two more changes (one on the high-level wording, turning this from "all pseudomodules" to "general pseudomodules" / pseudomodules don't affect a particular other module), and one picking yet another pseudomodule that *does* fall into that category. Please check if you still approve.

**Comment by kfessel** [comment]
> Force-pushed without modifications to get a CircleCI build of the documentation, to verify that it looks as nice in actual builds as it looks locally.

Your local build might be more represantative than the circle-ci result #17580

**Comment by chrysn** [comment]
Both look fine now, actually. Merging held until until either @fjmolinas reaffirms the ACK (eg. by merging it) or someone else also ACKs.

**Review by fjmolinas — COMMENTED** [review]
One comment I didn't see before..., you can squash it right away. But I would suggest doing so after the current build finishes and skipping ci build in the next build.

File: makefiles/pseudomodules.inc.mk [hunk 2]
```diff
@@ -1,6 +1,38 @@
...
+##
+## These are implemented in other modules or core components,
+## and serve to enable certain functionality.
+##
+## Here, pseudomodules are used instead of plain defines (that would be set using `CFLAGS += -DMODULE_NAME`)
+## because they can participate in dependency resolution:
+## they can pull in other modules.
+##
+## Pseudomodules are often enabled automatically through module dependencies,
+## but can also be enabled manually by stating `USEPKG += module_name` in the Makefile.
```
[the remark this answers is not in the document]
Comment (chrysn) [reply makefiles/pseudomodules.inc.mk]: Thanks. Applied, squashed, compared to the build-test-passing 9d6b2df8 showing only this change, and merging once all the rest of the checks is through.

**Pull request merged**